tools/clang/unittests/HLSL/ValidationTest.cpp

```diff
@@ -4207,6 +4210,74 @@ TEST_F(ValidationTest, ValidateWithHash) {
   VERIFY_ARE_EQUAL(memcmp(Result, pHeader->Hash.Digest, sizeof(Result)), 0);
 }
 
+// For now, 3 things are tested:
+// 1. The environment variable is not set. GetDxilDllPath() is empty and
+// DxilDllFailedToLoad() returns false
+// 2. Given a bogus path in the environment variable, GetDxilDllPath()
+// retrieves the path but fails to load it as a dll, and returns true
+// for DxilDllFailedToLoad()
+// 3. CLSID_DxcCompiler, CLSID_DxcLinker, CLSID_DxcValidator
+// may be created through DxcDllExtValidationSupport.
+// This is all to simply test that the new class, DxcDllExtValidationSupport,
+// works as intended.
+
+TEST_F(ValidationTest, UnitTestExtValidationSupport) {
+  DxcDllExtValidationSupport m_dllExtSupport1;
+  DxcDllExtValidationSupport m_dllExtSupport2;
+
+  // 1. with no env var set, test GetDxilDllPath() and DxilDllFailedToLoad()
```

Comment:
We should save/restore the current value for `DXC_DXIL_DLL_PATH`, and explicitly clear it for the no-path case.